assets/components/textInput/textInput.jsx

```diff
@@ -9,13 +9,30 @@ import { classNameWithModifiers } from 'helpers/utilities';
 
 // ----- Types ----- //
 
+type AutoComplete = 'on' | 'off';
+
+type AutoCapitalize = 'off'
+                    | 'none'
+                    | 'on'
+                    | 'sentences'
+                    | 'words';
+
+type InputType = 'text'
+               | 'email'
+               | 'password'
+               | 'url'
+               | 'tel';
+
 type PropTypes = {
   id: string,
+  type: InputType,
   labelText: string,
   onChange: (name: string) => void,
   value: string,
   placeholder: string,
   required: boolean,
+  autocomplete?: AutoComplete,
+  autocapitalize?: AutoCapitalize,
```

Comment:
Flow usually doesn't like `?` on props if they have defaults (see #702). However, it seems happy here 🤔.

Reply:
tbh, I have no idea how this works. initially I had nothing in `defaultProps` and it complained, so I thought ... 🙄 😅

Reply:
In summary, the problem we were having is that ESLint wants you to put `?` on parameters that have default props, but Flow doesn't. They were fighting each other so we decided to go with Flow and turn that particular ESLint rule off. So now, if you have a default parameter, you mark the prop as required and Flow figures out that it has a default from `defaultProps`.

The confusing thing is that Flow *isn't* complaining here, when it should be based upon this ^ 😔